src/utilities.jl

```diff
@@ -0,0 +1,59 @@
+_incomplete_job(job) = !(job["status"] in ["SUCCEEDED", "FAILED"])
+_suffix_asterisk(prefix) = endswith(prefix, "*") ? prefix : string(prefix, "*")
+
+function _get_job_ids(job_queue, prefix)
+    # Check if the prefix provided ends w/ '*", if not append it
+    prefix = _suffix_asterisk(prefix)
+
+    resp = @mock Batch.list_jobs(
+        Dict(
+            "jobQueue" => job_queue,
+            "filters" => [Dict("name" => "JOB_NAME", "values" => [prefix])],
+        ),
+    )
+    jobs = resp["jobSummaryList"]
+
+    while haskey(resp, "nextToken")
+        resp = @mock Batch.list_jobs(
+            Dict("jobQueue" => job_queue, "nextToken" => resp["nextToken"])
+        )
+        append!(jobs, resp["jobSummaryList"])
+    end
+
+    filter!(j -> _incomplete_job(j), jobs)
```

Comment:
Would it be better to use `_incomplete_job` in an if condition to only `append!` incomplete jobs as the list is being built instead of filtering after?

Reply:
I'm not sure if I'm setting this up the best way, however the reason why I did it like this is to avoid doing a filter w/ `_incomplete_job()` twice.

If you look a bit above we're defining `jobs = resp["jobSummaryList"]` before the `while` loop. Some of these jobs may have been completed ones.

I just wanted to get something quickly out the door here, that still made sense. If you have a suggestion on a better way I'd love to hear. :)

Reply:
Ah I misread the code. This makes sense then. 👍

Reply:
No worries! I feel like this sort of action where we query for some data, loop over until we get all the results based off of some token happens quite often and we should ideally make some wrapper function.